README.md

```diff
@@ -3,12 +3,32 @@
 # zigbee-herdsman-converters
 Collection of device converters to be used with zigbee-herdsman.
 
-# Contributing
-To get started clone this repo and install the requirements with `npm ci`. To add a new device, you'll need to add it's configuration to devices.js and potentially create a new converter for it. For more information see [this](http://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html) page.
+## Contributing
+See [Zigbee2MQTT how to support new devices](http://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html).
 
+## Submitting a pull request
 If you'd like to submit a pull request, you should run the following commands to ensure your changes will pass the tests:
 ```sh
 npm run lint
 npm test
 ```
+
 If any of those commands finish with an error your PR won't pass the tests and will likely be rejected.
+
+## Documentation of definition meta property
+- `configureKey`: required when a 'configure' is defined, this key is used by the application to determine if the content of the configure has been changed and thus needs to re-execute it. For a currently unsupported device you can set this to 1.
```

Comment:
I wonder, can we somehow hash the content of configure() and use that as dynamic key in the database?
If the hash changes, we need to reconfigure? No more messing with increasing ids. Out of scope of this though, just something that popped into my head.

Reply:
Also thought about this (not sure if possible) but there are also situations where we want to change the configure() but do not want it to be reapplied (common for sleeping end devices).

Reply:
I general doing a reconfigure with sleeping devices is tricky currently.

I think this wouldn’t be an issue if we just set a flag that it needs a reconfigure and execute it once the device checks in. That would also take the quick time event type situation away when trying to manually reconfigure a sleeping device that can’t easily be moved and is tricky to reach.

Reply:
But what e.g. for the sonoff case were it can only be configured after rejoining?

Reply:
The bind is the problem there yeah, reporting can be updates just fine.
I suppose we can introduce a `configureOnce` meta or something. Where it will not try to be smart. As those, should be the exception.

Reply:
Agree, its also the only device I know which has this problem. Note that reconfiguring is already retried when a Zigbee message is received from a device. Maybe we should just not do the `configure()` on startup for end devices (only when we receive a message from it). Found some resource for getting a method body: https://stackoverflow.com/questions/14885995/how-to-get-a-functionss-body-as-string . Anyway let's fix it in a different issue.

Reply:
Yeah as I said early, this is something for separate PR for sure.
I think for end devices we should indeed only do a reconfigure when we receive a message and (db saved hash != hash) && !meta.configureOnce, and after interviewing.